src/booking/containers/booking-home/booking-home.component.ts

```diff
@@ -3,72 +3,62 @@ import { FormBuilder, FormControl, FormGroup } from '@angular/forms';
+import { combineLatest, Observable, of, Subscription, } from 'rxjs';
 import { map } from 'rxjs/operators';
 import { AppConstants } from '../../../app/app.constants';
 import { SessionStorageService } from '../../../app/services/session-storage/session-storage.service';
 import { TaskListFilterComponent } from '../../../work-allocation-2/components';
-import { LocationDataService } from '../../../work-allocation-2/services';
 import { Booking, BookingNavigationEvent, BookingProcess } from '../../models';
 import { BookingService } from '../../services';
-
 @Component({
   selector: 'exui-booking-home',
   templateUrl: './booking-home.component.html'
 })
 export class BookingHomeComponent implements OnInit, OnDestroy {
 
   @Input() public bookingProcess: BookingProcess;
+  @Input() public userId: string;
   @Output() public bookingProcessChange = new EventEmitter<BookingProcess>();
   @Output() public eventTrigger = new EventEmitter();
 
   public bookingTypeForm: FormGroup;
   public existingBookings: Booking[];
   private bookings$: Observable<any[]> | any;
-  private locations$: Observable<any>[];
   private combineResult$: Observable<any[]> | any;
   private existingBookingsSubscription: Subscription;
   private refreshAssignmentsSubscription: Subscription;
 
   constructor(
     private readonly fb: FormBuilder,
     private readonly bookingService: BookingService,
-    private readonly locationService: LocationDataService,
     private readonly router: Router,
     private readonly sessionStorageService: SessionStorageService,
     private readonly windowService: WindowService,
-    private readonly featureToggleService: FeatureToggleService
+    private readonly featureToggleService: FeatureToggleService,
   ) { }
 
   public ngOnInit() {
     this.bookingTypeForm = this.fb.group({
       bookingType: new FormControl(null)
     });
 
-    this.existingBookingsSubscription = this.bookingService.getBookings().subscribe((arr) => {
-      this.locations$ = arr.bookings.map(
-        booking => this.locationService.getSpecificLocations([booking.base_location_id]).pipe(
-          map(location => {
-            return {
-              ...booking,
-              locationName: location[0] && location[0].site_name
-            };
-          }
-          ))
-      );
-
-      this.bookings$ = forkJoin(this.locations$);
-      this.combineResult$ = combineLatest([this.bookings$, this.featureToggleService.isEnabled(AppConstants.FEATURE_NAMES.booking)]);
-      this.combineResult$.pipe(map(([bookingResults, bookingFeatureToggle]) => {
-        this.existingBookings = bookingResults as any;
-        this.orderByCurrentThenFuture();
-        this.bookingProcess.selectedBookingLocationIds = bookingFeatureToggle ? (bookingResults as any ).filter(p => new Date().getTime() < new Date(p.beginTime).getTime()).sort(this.sortBookings).map(p => p.base_location_id) : null;
-      })).subscribe();
-    },
-    err => {
+    if (this.userId) {
```

Comment:
Surekha, I think this logic change seems fine but it may need re-tested because there was a ticket about it Ordering, disabling, and enabling buttons according to some logic ...  I suggest if possible implementing only get service without touching the logic

Reply:
@tarikdogan-hmcts as discussed, the get full location API is being called in node layer rather than the angular layer where the it making multiple calls to get the full list of locations